demisto_sdk/commands/format/update_integration.py

```diff
@@ -11,6 +11,7 @@
                                                           SKIP_RETURN_CODE,
                                                           SUCCESS_RETURN_CODE)
 from demisto_sdk.commands.format.update_generic_yml import BaseUpdateYML
+from demisto_sdk.commands.format.update_script import ScriptYMLFormat
```

Comment:
As both the `update_integration` and `update_script` extend the update_generic_yml(BaseUpdateYML)
I think it is better to move the implementation of update_docker_image_in_script there to not create this dependency.

Reply:
I felt that it is less fitting as other yml formatters also extend BaseUpdateYML and the logic there should be generic to yml stuff and not scripts/integrations.

Reply:
I think otherwise, but understand it is not a decisive matter.

Reply:
I agree with @bakatzir here - we have some functions in the BaseUpdateYML that are used only by part of the classes inheriting from it (i.e. not part of the` update_yml` generic function).

For instance `update_tests` - is used by playbook formatter but not the test playbook formatter.

I believe this is the same case here as it might be confusing why should the docker be checked through the unrelated script formatter.

Reply:
@roysagi not sure I am fully following. I'll merge and feel free to do later on any design changes as you see fitting.